src/middlewared/middlewared/utils/io.py

```diff
@@ -8,7 +8,7 @@ def write_if_changed(path, data):
 
     changed = False
 
-    with open(os.open(path, os.O_CREAT | os.O_RDWR), 'wb+') as f:
+    with open(path, 'wb+') as f:
```

Comment:
I remember there was a good reason to use os.open but now I don't remember why. Yay for comments in the code.

Have you tested this extensively?

Reply:
I think the reason was because without os.open we couldnt actually READ *and* WRITE.
But could be a python bug at the time.

Reply:
@themylogin do you remember the exact reason?

Reply:
Yeah, I think it was probably a python bug at the time. `wb+` should open for read and write.

Reply:
I'm actually waiting to build a custom image with these changes (and the PR from the build repo) so I can test more thoroughly.

Reply:
> @themylogin do you remember the exact reason?

open(path, "wb+") includes `os.O_TRUNC`.

Reply:
Thanks @chris3torek ! Glad you're still around checking our mistakes, hope you + family are doing OK ; )

Reply:
Fixed 2 issues.
- specified mode
- changed to `rb+` to not cause confusion on what we're actually trying to do